src/treegen.cpp

```diff
@@ -111,9 +111,23 @@ void make_tree(MMVManip &vmanip, v3s16 p0,
 }
 
 
+static void set_param_to_id(INodeDefManager *ndef, MapNode *n, u8 id)
+{
+	content_t t = n->getContent();
+	const ContentFeatures &f = ndef->get(t);
+	if (f.param_type_2 == CPT2_NONE) {
```

Comment:
Note to myself: [suggestions for the id](http://irc.minetest.ru/minetest-dev/2015-07-18#i_4329210).